server/api/attendance/attendance.controller.js

```diff
@@ -335,17 +335,13 @@ exports.attend = function(req,res){
       }
       // Classyear attendance code and bonus code was incorrect, try small group
       else{
-        // if the user has no smallgroup, they cannont submit smallgroup attendance
-        if (!user.smallgroup || user.smallgroup === undefined || user.smallgroup === null){
-          return res.send(400, 'No small group found/incorrect daycode!');
-        }
-
-        return SmallGroup.findById(user.smallgroup, function(err, smallgroup){
+        return SmallGroup.findOne({"students":user._id, "classYear":classYear._id}, function(err, smallgroup){
           if (err) {return handleError(err)}
           // if the user has no smallgroup, they cannont submit smallgroup  attendance
-          if (!smallgroup){return res.send(400, 'Incorrect day code');}
+          if (!smallgroup){return res.send(400, 'No small group found or incorrect daycode!');}
```

Comment:
i think we are autoadding users to the smallgroup if they aren't part of one and they submit the right daycode. So Incorrect Daycode should be the message shouldn't it?